sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/InternalRow.scala

```diff
@@ -140,7 +140,7 @@ object InternalRow {
         case PhysicalLongType => (input, ordinal) => input.getLong(ordinal)
         case PhysicalFloatType => (input, ordinal) => input.getFloat(ordinal)
         case PhysicalDoubleType => (input, ordinal) => input.getDouble(ordinal)
-        case PhysicalStringType => (input, ordinal) => input.getUTF8String(ordinal)
+        case PhysicalStringType(_) => (input, ordinal) => input.getUTF8String(ordinal)
```

Comment:
It is better to don't depend on the number of fields, see https://github.com/databricks/scala-style-guide?tab=readme-ov-file#pattern-matching
```suggestion
        case _: PhysicalStringType => (input, ordinal) => input.getUTF8String(ordinal)
```

Reply:
Yep, makes sense. Note that collationId will be used in those expressions after the next PR :) But yeah, let's remove expansion until then.